OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -63,6 +63,8 @@ public class ArmamentInfo : ConditionalTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		public readonly bool ShootAtClosestTargetablePosition = false;
```

Comment:
Im actually in favor of B. Re-testing all the unit ranges shouldn't be a problem really as it makes logical sense to be able to hit a structure at the side that is visible. Also helps for units that have splash damage effects to have the full damage rather then partial.

The only issue I can see is artillery (Both TD and RA) and they would inadvertently become more accurate with on hits. Also would need testing with defense structures.

Does this work with commandos/tanya/engineers?